detectable/src/main/java/com/synopsys/integration/detectable/factory/DetectableFactory.java

```diff
@@ -889,6 +901,12 @@ private PipenvExtractor pipenvExtractor() {
         return new PipenvExtractor(executableRunner, pipenvTransformer(), pipenvFreezeParser(), pipenvJsonGraphParser());
     }
 
+    private PipfileLockExtractor pipfileLockExtractor() {return new PipfileLockExtractor(gson, pipfileLockParser(), pipfileLockTransformer());}
+
+    private PipfileLockParser pipfileLockParser() {return new PipfileLockParser();}
+
+    private PipfileLockDependencyTransformer pipfileLockTransformer() {return new PipfileLockDependencyTransformer();}
```

Comment:
We aren't creating separate methods for each component anymore unless the component is shared by multiple detectors.

Check out `DetectableFactory::createPearCliDetectable` for a good example of how to construct a Detectable and use the Options file.